core/src/main/java/org/apache/accumulo/core/conf/Property.java

```diff
@@ -746,6 +746,8 @@ public enum Property {
       "The number of threads used to delete RFiles and write-ahead logs", "1.3.5"),
   GC_TRASH_IGNORE("gc.trash.ignore", "false", PropertyType.BOOLEAN,
       "Do not use the Trash, even if it is configured.", "1.5.0"),
+  GC_TRASH_IGNORE_IMPORTS_ONLY("gc.trash.ignore.imports.only", "false", PropertyType.BOOLEAN,
+      "Skip trash only for Import files when gc.trash.ignore is true.", "3.0.0"),
```

Comment:
I find this confusing,  Just from the descriptions, if the system is configured to use trash - then setting GC_TRASH_IGNORE = true overrides the system setting and files are not moved to trash and are immediately deleted.

With GC_TRASH_IGNORE=true, setting GC_TRASH_IGNORE_IMPORTS_ONLY=true - then bulk import files skip the trash, but all other files will use the trash?

Would it be easier for users if these were more independent?

- GC_TRASH_IGNORE true - skip trash, false - use trash if configured for the system.
- GC_TRASH_IGNORE_IMPORTS_ONLY - true - skip trash for bulk import files, overrides any GC_TRASH_IGNORE setting for bulk import files. false - use GC_TRASH_IGNORE or the system setting.

My thinking is along the lines of:

| system        | GC_TRASH_IGNORE | GC_TRASH_IGNORE_IMPORTS_ONLY | impact                |
|---------------|-----------------|------------------------------|-----------------------|
| trash disable | n/a             | n/a                          | skip trash, all files |
| trash enabled | N               | N                            | ues trash, all files  |
| trash enabled | N               | Y  | skip trash, bulk only |
| trash enabled | Y  | N | skip trash, all files |
| trash enabled | Y  | Y  | skip trash, all files |

Reply:
@alerman and I had a discussion about this. He suggested renaming the properties to `GC_USE_TRASH`. It was very confusing to me working on this because things were not named in the affirmative, there were a bunch of negations in the code.

Reply:
That's what I ran into too - but renaming the GC_TRASH_IGNORE property seemed something to be avoided. Maybe if the new property was `GC_TRASH_IGNORE_IMPORTS` - that would make having the usage I described above easier to communicate?

Reply:
```
  GC_USE_TRASH("gc.trash.ignore", "true", PropertyType.BOOLEAN,
      "If trash is configured, move deleted non-bulk import files there instead of deleting them", "1.5.0"),
  GC_BULK_IMPORT_USE_TRASH("gc.trash.use.bulk.imports", "true", PropertyType.BOOLEAN,
      "If trash is configured, move deleted bulk import files there instead of deleting them", "3.0.0"),
```

| Trash Enabled | GC_USE_TRASH | GC_BULK_IMPORT_USE_TRASH | Result |
|----------------------|------------------------|-----------------------------------------------|-----------|
| N | N/A | N/A | all files deleted |
| Y | true | true | all files moved to trash |
| Y | false | true | only bulk import files moved to trash |
| Y | true | false | bulk import files deleted , others moved to trash |
| Y | false | false | all files deleted |

Reply:
That's fine too - but that requires property upgrade code to translate the settings - and may have issues because we may need to deprecate, but keep GC_TRASH_IGNORE? Not sure...

Reply:
The table helps, but it also highlights how unintuitive having three separate properties are to control a desired outcome. I would much rather have a simple single property that had multiple options to select the specific policy, rather than leave it as an exercise for the user to try to determine the current policy by looking it up in a table like this.